rust/email/src/lib.rs

```diff
@@ -36,20 +39,36 @@ fn c_char_to_string(str_: *const c_char) -> Result<String, i32> {
 	}
 }
 
+/// Reads the email body from a file, located in `filepath`.
+///
+/// # Arguments
+///
+/// * `filepath` - A String containing the filepath of the file
+///   we want to read.
+fn read_mail_body(filepath: String) -> String {
+	let body = fs::read_to_string(filepath)
+		.expect("Couldn't read file");
+
+	return body;
+}
```

Comment:
Some requests:
* Argument should be `filepath: &str` to avoid ownership transfer and unnecessary cloning.
* Instead of using `.expect()`, it's often better to return a `Result` type, like `Result<String, io::Error>` in our case. This will allow the function to just contain `fs::read_to_string(filepath)` and match the result later in the `send_email` function, giving an error message to the user.

Reply:
Good idea about returning the `Result`. I was thinking of implementing it like that, but didn't know if it would work for you. I am on it.